### ISSUE TITLE
Suggestion: Add zopfli as second deflate-option

### ISSUE THREAD

**Issue by mkver** [mbunkus/mkvtoolnix#1708]
Hello,

[zopfli](https://github.com/google/zopfli) is another implementation of the deflate-compression; it is [slower, but more size-efficient](https://zopfli.googlecode.com/files/Data_compression_using_Zopfli.pdf) than other deflate algorithms like zlib, the currently used one. But importantly the decompression time seems to be unaffected by this (see table 3 of the pdf above) so that there will be no playback issues. According to the description they are compatible with zlib (they can create both deflate and zlib streams in memory). And it is a free and open-source project and so it should be compatible with MkvToolNix.
Given that matroska-files are usually only muxed once and up- and downloaded a lot, it makes sense to invest more time into muxing.

Grüße
Andi

**Comment by mbunkus** [comment]
Thanks for the suggestion. Sounds somewhat useful (though we're talking about saving maybe a few KB in GB sized files). I'll think about it.

**Comment by mbunkus** [comment]
I've decided against adding support for Zopfli. I've made a simple test by replacing the current ZLIB compression method with calls to Zopfli. Muxing a file with eight VobSub tracks resulting in a 4,672,546 bytes long ZLIB compressed file takes less than a second with ZLIB. With Zopfli (even with only five iterations) the time went up to while only saving 40,568 bytes (0.86%).

As mkvmerge isn't multi-threaded such an increase would simply by on top of the time required for all those I/O operations (which is what usually limits mkvmerge's speed). I don't see a lot of value for end users in offering such a method, not even an optional ones. Sorry.
